dind/docker-compose.yml

```diff
@@ -0,0 +1,25 @@
+version: '3'
+
+services:
+
+    node1:
+        image: anybox/cluster_node_dind
```

Comment:
As you provide a Dockerfile, instead using an image from a registry or build earlier in shell script you could ask docker-compose to build the image while starting up the service, using [build directive](https://docs.docker.com/compose/compose-file/#build).